### PR TITLE
tranform discogs artist name also on fetch artist by id

### PR DESCRIPTION
**Pull request by DanielW1987**

### CODE REVIEW

**Comment by DanielW1987** [comment]
[saveAndFetchArtist has several bugs](https://trello.com/c/VxMc1Dtb/205-saveandfetchartist-has-several-bugs)

File: discogs/src/test/java/rocks/metaldetector/discogs/client/transformer/DiscogsArtistSearchResultContainerTransformerTest.java [hunk 1]
```diff
@@ -13,9 +17,19 @@
 
 import java.util.List;
 
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@ExtendWith(MockitoExtension.class)
 class DiscogsArtistSearchResultContainerTransformerTest implements WithAssertions {
```
Comment (hedgehog1833) [inline discogs/src/test/java/rocks/metaldetector/discogs/client/transformer/DiscogsArtistSearchResultContainerTransformerTest.java:26]: Please reset the mock in @AfterEach

File: discogs/src/test/java/rocks/metaldetector/discogs/client/transformer/DiscogsArtistTransformerTest.java [hunk 2]
```diff
@@ -16,9 +20,19 @@
 import java.util.List;
 import java.util.stream.Stream;
 
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@ExtendWith(MockitoExtension.class)
 class DiscogsArtistTransformerTest implements WithAssertions {
```
Comment (hedgehog1833) [inline discogs/src/test/java/rocks/metaldetector/discogs/client/transformer/DiscogsArtistTransformerTest.java:29]: Same here... I'm starting to wonder 🧐  Didn't we always reset our mocks?^^

File: discogs/src/test/java/rocks/metaldetector/discogs/config/DiscogsRequestInterceptorTest.java [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+  @DisplayName("should set accept header always to 'application/json'")
+  void accept_header_is_set() throws IOException {
+    // given
+    doReturn(httpHeadersMock).when(httpRequestMock).getHeaders();
+
+    // when
+    underTest.intercept(httpRequestMock, new byte[]{}, clientHttpRequestExecutionMock);
+
+    // then
+    verify(httpHeadersMock, times(1)).setAccept(List.of(MediaType.APPLICATION_JSON));
```
Comment (hedgehog1833) [inline discogs/src/test/java/rocks/metaldetector/discogs/config/DiscogsRequestInterceptorTest.java:51]: Static import?

File: discogs/src/test/java/rocks/metaldetector/discogs/config/DiscogsRequestInterceptorTest.java [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+    // given
+    var userAgent = "User Agent";
+    doReturn(httpHeadersMock).when(httpRequestMock).getHeaders();
+    doReturn(userAgent).when(discogsConfig).getUserAgent();
+
+    // when
+    underTest.intercept(httpRequestMock, new byte[]{}, clientHttpRequestExecutionMock);
+
+    // then
+    verify(httpHeadersMock, times(1)).set(HttpHeaders.USER_AGENT, userAgent);
```
Comment (hedgehog1833) [inline discogs/src/test/java/rocks/metaldetector/discogs/config/DiscogsRequestInterceptorTest.java:66]: Static import?

File: discogs/src/test/java/rocks/metaldetector/discogs/config/DiscogsRequestInterceptorTest.java [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    // given
+    var token = "discogs token";
+    doReturn(httpHeadersMock).when(httpRequestMock).getHeaders();
+    doReturn(token).when(discogsConfig).getAccessToken();
+
+    // when
+    underTest.intercept(httpRequestMock, new byte[]{}, clientHttpRequestExecutionMock);
+
+    // then
+    verify(httpHeadersMock, times(1)).set(HttpHeaders.AUTHORIZATION, TOKEN_PREFIX + token);
```
Comment (hedgehog1833) [inline discogs/src/test/java/rocks/metaldetector/discogs/config/DiscogsRequestInterceptorTest.java:81]: Static import?

**Pull request merged**